drivers/ieee802154/ieee802154_rf2xx.h

```diff
@@ -0,0 +1,119 @@
+/* Runtime context structure
+ ***************************
+ */
+typedef enum rf2xx_trx_state_cmd_tag {
+	RF2XX_TRX_PHY_STATE_CMD_NOP             = 0x00,
+	RF2XX_TRX_PHY_STATE_CMD_TX_START        = 0x02,
+	RF2XX_TRX_PHY_STATE_CMD_FORCE_TRX_OFF   = 0x03,
+	RF2XX_TRX_PHY_STATE_CMD_FORCE_PLL_ON    = 0x04,
+	RF2XX_TRX_PHY_STATE_CMD_RX_ON           = 0x06,
+	RF2XX_TRX_PHY_STATE_CMD_TRX_OFF         = 0x08,
+	RF2XX_TRX_PHY_STATE_CMD_PLL_ON          = 0x09,
+	RF2XX_TRX_PHY_STATE_CMD_PREP_DEEP_SLEEP = 0x10,
+	RF2XX_TRX_PHY_STATE_CMD_RX_AACK_ON      = 0x16,
+	RF2XX_TRX_PHY_STATE_CMD_TX_ARET_ON      = 0x19,
+	/* Implemented by Software */
+	RF2XX_TRX_PHY_STATE_CMD_SLEEP           = 0x0F,
+	RF2XX_TRX_PHY_STATE_CMD_DEEP_SLEEP      = 0x20,
+} rf2xx_trx_state_cmd_t;
+
+typedef enum rf2xx_trx_state_trac_tag {                        /* RX_AACK TX_ARET  */
+	RF2XX_TRX_PHY_STATE_TRAC_SUCCESS                = 0x00, /*    X       X     */
+	RF2XX_TRX_PHY_STATE_TRAC_SUCCESS_DATA_PENDING   = 0x01, /*            X     */
+	RF2XX_TRX_PHY_STATE_TRAC_SUCCESS_WAIT_FOR_ACK   = 0x02, /*    X             */
+	RF2XX_TRX_PHY_STATE_TRAC_CHANNEL_ACCESS_FAILED  = 0x03, /*            X     */
+	RF2XX_TRX_PHY_STATE_TRAC_NO_ACK                 = 0x05, /*            X     */
+	RF2XX_TRX_PHY_STATE_TRAC_INVALID                = 0x07, /*    X       X     */
+} rf2xx_trx_state_trac_t;
+
+typedef enum rf2xx_trx_state_tag {
+	RF2XX_TRX_PHY_STATE_INITIAL,
+	RF2XX_TRX_PHY_STATE_IDLE,
+	RF2XX_TRX_PHY_STATE_SLEEP,
+	RF2XX_TRX_PHY_BUSY_RX,
+	RF2XX_TRX_PHY_BUSY_TX,
+} rf2xx_trx_state_t;
+
+typedef enum rf2xx_trx_model_tag {
+	RF2XX_TRX_MODEL_INV     = 0x00,
+	RF2XX_TRX_MODEL_230     = 0x02,
+	RF2XX_TRX_MODEL_231     = 0x03,
+	RF2XX_TRX_MODEL_212     = 0x07,
+	RF2XX_TRX_MODEL_232     = 0x0A,
+	RF2XX_TRX_MODEL_233     = 0x0B,
+} rf2xx_trx_model_t;
+
+typedef struct rf2xx_dt_gpio_tag {
+	const char *devname;
+	u32_t pin;
+	u32_t flags;
+} rf2xx_dt_gpio_t;
+
+typedef struct rf2xx_dt_spi_tag {
+	const char *devname;
+	u32_t freq;
+	u32_t addr;
+	rf2xx_dt_gpio_t cs;
+} rf2xx_dt_spi_t;
+
+struct rf2xx_config {
+	u8_t inst;
+	u8_t hwcaps;
+
+	rf2xx_dt_gpio_t irq;
+	rf2xx_dt_gpio_t reset;
+	rf2xx_dt_gpio_t slptr;
+	rf2xx_dt_gpio_t dig2;
+	rf2xx_dt_gpio_t clkm;
+
+	rf2xx_dt_spi_t spi;
+};
+
+struct rf2xx_context {
+	struct net_if *iface;
+	/**************************/
+	struct device *irq_gpio;
+	struct device *reset_gpio;
+	struct device *slptr_gpio;
+	struct device *dig2_gpio;
+	struct device *clkm_gpio;
+	struct gpio_callback irq_cb;
+	struct device *spi;
+	struct spi_config spi_cfg;
+	struct spi_cs_control spi_cs;
+	u8_t mac_addr[8];
+	/************RF************/
+	rf2xx_trx_model_t trx_model;
+	u8_t trx_version;
+	s8_t trx_rssi_base;
+	s8_t trx_rssi;
+	u8_t trx_lqi;
+	u8_t trx_ed;
+	rf2xx_trx_state_t trx_state;
+	rf2xx_trx_state_trac_t trx_trac;
+	bool trx_rx_state;
+	struct k_thread trx_thread;
+	K_THREAD_STACK_MEMBER(trx_stack,
+		CONFIG_IEEE802154_RF2XX_RX_STACK_SIZE);
+	struct k_sem trx_isr_lock;
```

Comment:
naming issue? that lock is given by txr_isr_handler (it's confusing txr/trx)

Reply:
Done.

Reply:
Good Point! Fixed ISR name.